src/simstation/Agent.java

```diff
@@ -1,13 +1,122 @@
 package simstation;
 
-public class Agent {
+// Assuming Heading is an enum already defined somewhere
+import mvc.Model;
+import mvc.Utilities;
 
+import java.io.Serializable;
+
+public abstract class Agent implements Runnable, Serializable {
+
+    private String name;
     protected Heading heading;
-    public Agent() {
+    public int xc, yc; // Presumably for coordinates, but not shown in your diagram
+    private boolean suspended = false;
+    private boolean stopped = false;
+    private transient Thread myThread; // Transient because Threads are not serializable
+    public Simulation SimStation;
```

Comment:
we need to remove the SimStation variable cause in the random walk package he provided the constructor of drunk does not take in a model

Reply:
also look for a comment I make below about modifying some of the other functions to facilitate this